packager/launchEditor.js

```diff
@@ -30,19 +50,22 @@ function launchEditor(fileName, lineNumber) {
     return;
   }
 
-  var argument = fileName;
+  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
+  var args = [fileName];
   if (lineNumber) {
-    argument += ':' + lineNumber;
+    args = getArgumentsForLineNumber(editor, fileName, lineNumber);
   }
 
-  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
   if (editor) {
     console.log('Opening ' + chalk.underline(fileName) + ' with ' + chalk.bold(editor));
-    exec(editor + ' ' + argument, function(error) {
-      if (error) {
-        console.log(chalk.red(error.message));
-        printInstructions('How to fix');
+
+    var child = spawn(editor, args, {stdio: 'inherit'});
```

Comment:
Yeah it's for terminal editors @ide -- otherwise the process starts for those editors but never attaches to anything (which might be confusing for devs).

If we don't want to support terminal editors we could switch to just escaping the filepath but we should error out if your editor is anything besides atom / sublime / mvim

Sent from my iPhone

On Jul 11, 2015, at 5:34 PM, James Ide <notifications@github.com<mailto:notifications@github.com>> wrote:


In packager/launchEditor.js<https://github.com/facebook/react-native/pull/1957#discussion_r34417537>:

>    if (editor) {
>      console.log('Opening ' + chalk.underline(fileName) + ' with ' + chalk.bold(editor));
> -    exec(editor + ' ' + argument, function(error) {
> -      if (error) {
> -        console.log(chalk.red(error.message));
> -        printInstructions('How to fix');
> +
> +    var child = spawn(editor, args, {stdio: 'inherit'});


This is for terminal editors? They might cause a mess since the packager could keep writing to stdout while the editor is trying to write to stdout too, although maybe it's fine in practice.

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/react-native/pull/1957/files#r34417537>.